jmbase/jmbase/support.py

```diff
@@ -1,16 +1,52 @@
 from __future__ import (absolute_import, division,
                         print_function, unicode_literals)
 from builtins import * # noqa: F401
-from future.utils import iteritems
-
-import sys
 
 import logging
-import pprint
 from getpass import getpass
 
-logFormatter = logging.Formatter(
-    "%(asctime)s [%(threadName)-12.12s] [%(levelname)-5.5s]  %(message)s")
+from chromalog.log import (
+    ColorizingStreamHandler,
+    ColorizingFormatter,
+)
+from chromalog.colorizer import GenericColorizer
+from colorama import Fore, Back, Style
+
+# magic; importing e.g. 'info' actually instantiates
+# that as a function that uses the color map
+# defined below. ( noqa because flake doesn't understand)
+from chromalog.mark.helpers.simple import (  # noqa: F401
+    debug,
+    info,
+    important,
+    success,
+    warning,
+    error,
+    critical,
+)
+
+# our chosen colorings for log messages in JM:
+jm_color_map = {
+    'debug': (Style.DIM + Fore.LIGHTBLUE_EX, Style.RESET_ALL),
+    'info': (Style.BRIGHT + Fore.BLUE, Style.RESET_ALL),
+    'important': (Style.BRIGHT, Style.RESET_ALL),
+    'success': (Fore.GREEN, Style.RESET_ALL),
+    'warning': (Fore.YELLOW, Style.RESET_ALL),
+    'error': (Fore.RED, Style.RESET_ALL),
+    'critical': (Back.RED, Style.RESET_ALL),
+}
+
+class JMColorizer(GenericColorizer):
+    default_color_map = jm_color_map
+
+jm_colorizer = JMColorizer()
+
+# Sanity test that coloring is working on startup
+print(jm_colorizer.colorize_message(
```

Comment:
Having a `print` on the top level seems weird. Is this a testing leftover?

Reply:
As per line 44 comment, it's just a "starting up, logging (colored) is working" message. It could be rewritten to use jmprint() (see other comment thread), but I don't think it really matters.

Reply:
when importing a package usually the last thing you want to have happening is random console output.

Reply:
Ah yes, I get your point. Agreed, will remove.